esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3769,6 +3769,11 @@ void WsWuHelpers::submitWsWorkunit(IEspContext& context, IConstWorkUnit* cw, con
         }
     }
 
+    ISpan * activeSpan = context.queryActiveSpan();
```

Comment:
@afishbeck if you submitted a workunit the globalids etc. where not preserved in the workunit.  They are now.  Should this code be commoned up with the code in ws_ecl_service.cpp?

Reply:
Yes, when ws_ecl_service.cpp runs the job as a workunit (as opposed to calling out to roxie) it clones the original workunit.  So submitWsEclWorkunit is a bit different BUT it has context available so should be able to be commoned up with this.